BM/instruction-check/feature_list.py

```diff
@@ -1,51 +1,228 @@
+    },
+    "GFNI": {
+        "cpuid": ['7', '0', '0', '0', 'c', '8'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "VAES": {
+        "cpuid": ['7', '0', '0', '0', 'c', '9'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "AVX512_VNNI": {
+        "cpuid": ['7', '0', '0', '0', 'c', '11'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "AVX512_BITALG": {
+        "cpuid": ['7', '0', '0', '0', 'c', '12'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "RDPID": {
+        "cpuid": ['7', '0', '0', '0', 'c', '22'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "KL": {
+        "cpuid": ['7', '0', '0', '0', 'c', '23'],
+        "platforms": {}
+    },
+    "CLDEMOTE": {
+        "cpuid": ['7', '0', '0', '0', 'c', '25'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "MOVDIRI": {
+        "cpuid": ['7', '0', '0', '0', 'c', '27'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "MOVDIR64B": {
+        "cpuid": ['7', '0', '0', '0', 'c', '28'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "PKS": {
+        "cpuid": ['7', '0', '0', '0', 'c', '31'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "UINTR": {
+        "cpuid": ['7', '0', '0', '0', 'd', '5'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "AVX512_VP2INTERSECT": {
+        "cpuid": ['7', '0', '0', '0', 'd', '8'],
+        "platforms": {}
+    },
+    "SERIALIZE": {
+        "cpuid": ['7', '0', '0', '0', 'd', '14'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "TSXLDTRK": {
+        "cpuid": ['7', '0', '0', '0', 'd', '16'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "CET_IBT": {
+        "cpuid": ['7', '0', '0', '0', 'd', '20'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "AMX_BF16": {
+        "cpuid": ['7', '0', '0', '0', 'd', '22'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "AVX512_FP16": {
+        "cpuid": ['7', '0', '0', '0', 'd', '23'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "AMX_TILE": {
+        "cpuid": ['7', '0', '0', '0', 'd', '24'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "AMX_INT8": {
+        "cpuid": ['7', '0', '0', '0', 'd', '25'],
+        "platforms": {"SPR", "EMR", "GNR"}
+    },
+    "AVX_VNNI": {
+        "cpuid": ['7', '0', '1', '0', 'a', '4'],
+        "platforms": {"SPR", "EMR", "GNR", "SRF", "CWF"}
+    },
+    "CMPCCXADD": {
+        "cpuid": ['7', '0', '1', '0', 'a', '7'],
+        "platforms": {"SRF", "CWF"}
+    },
+    "FRED": {
+        "cpuid": ['7', '0', '1', '0', 'a', '17'],
+        "platforms": {}
+    },
+    "WRMSRNS": {
+        "cpuid": ['7', '0', '1', '0', 'a', '19'],
+        "platforms": {"SRF", "CWF"}
+    },
+    "AMX_FP16": {
+        "cpuid": ['7', '0', '1', '0', 'a', '21'],
+        "platforms": {"SPR", "EMR", "GNR"}
```

Comment:
Seems SPR/EMR doesn't support AMX_FP16. 

Did you test your patch on these platform?

Reply:
Thank you for your suggestion. I previously conducted tests on SPR, GNR, and SRF, but there may be some errors. I will double-check.